src/main/webapp/app/shared/components/frontend-search/frontend-search.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+import { Subject } from 'rxjs';
+import { debounce, takeUntil } from 'rxjs/operators';
+import { DEFAULT_DEBOUNCE } from 'app/app.constants';
+
+@Component({
+  selector: 'app-frontend-search',
+  templateUrl: './frontend-search.component.html',
+  styleUrls: ['./frontend-search.component.scss'],
+})
+export class FrontendSearchComponent implements OnInit, OnDestroy {
+  @Input()
+  placeholder?: string;
+
+  @Output()
+  valueChange = new EventEmitter<string>();
+
+  searchForm: FormControl = new FormControl();
```

Comment:
rename component: `clientside-filter.component.ts`.
property: `searchControl` look for implicity.